lang/cs/Org.Apache.REEF.IMRU/OnREEF/Driver/ServiceAndContextConfigurationProvider.cs

```diff
@@ -138,32 +157,32 @@ internal ContextAndServiceConfiguration GetDataLoadingConfigurationForEvaluatorB
             }
 
             Logger.Log(Level.Info, "Getting a new data loading configuration");
-            _partitionIdProvider[evaluatorId] = _partitionDescriptorIds.Pop();
+            _partitionContextIdProvider[evaluatorId] = _avilablePartitionDescriptorContextIds.Pop();
```

Comment:
Wouldn't it be better to use
```C#
_partitionContextIdProvider.Add(evaluatorId, _avilablePartitionDescriptorContextIds.Pop());
```
and let it raise `ArgumentException` (thrown from `Add`) and `InvalidOperationException` (thrown from `Pop`) and then handle those exceptions instead of writing all those `if` statements upfront?